crates/tabby/src/services/chat.rs

```diff
@@ -85,3 +86,23 @@ pub async fn create_chat_service(model: &str, device: &Device, parallelism: u8)
 
     ChatService::new(engine, chat_template)
 }
+
+/// Format a [ChatCompletionChunk] into an OpenAI-compatible JSON event
+pub fn format_chunk_to_event(
+    chunk: ChatCompletionChunk,
```

Comment:
Modify the definition of ChatCompletionChunk to match openai's schema definition, and update logic of [ChatService::generate](https://github.com/TabbyML/tabby/blob/4a02a316be4247a3c582b6ee75865cf9ee67b209/crates/tabby/src/services/chat.rs#L70)